Http/Http.cs

```diff
@@ -0,0 +1,352 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace DUCK.Http
+{
+	// How to use
+	// First use one of Http static helper methods to create a HttpRequest. for example we will do a get request.
+	// Http.Get("uri");
+	// This will return a HttpRequest setup for Http Verb GET.
+	// The next part is to actually send the HttpRequest, we do this by simply calling Send() on the HttpRequest.
+	// Http.Get("uri").Send();
+	// The Send method signature has the options for onSuccess and onError callbacks.
+	// HttpRequest.Send(Action<HttpResponse> onSuccess = null, Action<HttpResponse> onError = null);
+	// Both callbacks include the HttpResponse object.
+	// You can get the Body of the response in 4 different forms: Text, Bytes, Texture or as an object parsed from Json.
+	// Heres what it looks like parsed to an object:
+	// Http.Get("uri").Send(response => { response.ParseBodyAs<User>(); });
+	//
+	// Alternatively you can use Http for sending UnityWebRequest with callbacks. For example:
+	// Http.Instance.Send(UnityWebRequest.Post("uri", "payload"), (UnityWebRequest request) => { });
+	// Http.Instance.Send(UnityWebRequest.Post("uri", "payload"), onError: () => { });
+	// Http.Instance.Send(UnityWebRequest.Post("uri", "payload"), onNetworkError: () => { });
+	public sealed class Http : MonoBehaviour
+	{
+		private static Http instance;
+
+		public static Http Instance
+		{
+			get
+			{
+				if (instance == null)
+				{
+					instance = new GameObject("Http").AddComponent<Http>();
+				}
+
+				return instance;
+			}
+		}
+
+		private Dictionary<string, string> superHeaders;
+
+		private void Awake()
+		{
+			instance = this;
+			instance.hideFlags = HideFlags.HideInHierarchy;
+
+			superHeaders = new Dictionary<string, string>();
+		}
+
+		/// <summary>
+		/// Super headers are key value pairs that will be added to every subsequent HttpRequest.
+		/// </summary>
+		/// <returns></returns>
+		public Dictionary<string, string> GetSuperHeaders()
+		{
+			return superHeaders;
+		}
+
+		/// <summary>
+		/// Sets a header to the SuperHeaders key value pair, if the header key already exists, the value will be replaced.
+		/// </summary>
+		/// <param name="key">The header key to be set.</param>
+		/// <param name="value">The header value to be assigned.</param>
+		public void SetSuperHeader(string key, string value)
+		{
+			superHeaders[key] = value;
+		}
+
+		/// <summary>
+		/// Removes a header from the SuperHeaders list.
+		/// </summary>
+		/// <param name="key">The header key to be removed.</param>
+		/// <returns></returns>
```

Comment:
This returns comment needs filling in saying it returns whether the key was removed successfully